### PR TITLE
Define "HAVE_UNISTD_H" in zlib on Mac systems (#2585)

### LINKED ISSUE #2585: MacOS build is failing
**Describe the bug**
GitHub MacOS build is failing because zlib build is using C99 for some reason, since the recent upgrade from 1.2.11 to 1.2.12.

Can we upgrade to C11?

Alternatively we could apply `-Wimplicit-function-declaration` to MacOS premake, although that is less ideal.

What about this: https://stackoverflow.com/questions/14228856/how-to-compile-c-with-c11-support-in-mac-terminal

**Version**
Since https://github.com/multitheftauto/mtasa-blue/commit/30c68a8754f87cdb0fa13c20e1cf6e6038a498b3

**Additional context**
https://github.com/multitheftauto/mtasa-blue/runs/5775453047?check_suite_focus=true#step:6:775
```
==== Building zlib (debug_x64) ====
Creating obj/x64/Debug/zlib
adler32.c
compress.c
crc32.c
deflate.c
gzclose.c
gzlib.c
../vendor/zlib/gzlib.c:254:9: error: implicit declaration of function 'lseek' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
        LSEEK(state->fd, 0, SEEK_END);  /* so gzoffset() is correct */
gzread.c
        ^
../vendor/zlib/gzlib.c:14:17: note: expanded from macro 'LSEEK'
#  define LSEEK lseek
                ^
../vendor/zlib/gzlib.c:254:9: note: did you mean 'fseek'?
../vendor/zlib/gzlib.c:14:17: note: expanded from macro 'LSEEK'
#  define LSEEK lseek
                ^
/Applications/Xcode_13.2.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/stdio.h:162:6: note: 'fseek' declared here
int      fseek(FILE *, long, int);
         ^
../vendor/zlib/gzlib.c:260:24: error: implicit declaration of function 'lseek' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
        state->start = LSEEK(state->fd, 0, SEEK_CUR);
                       ^
../vendor/zlib/gzlib.c:14:17: note: expanded from macro 'LSEEK'
#  define LSEEK lseek
                ^
../vendor/zlib/gzlib.c:361:9: error: implicit declaration of function 'lseek' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
    if (LSEEK(state->fd, state->start, SEEK_SET) == -1)
        ^
../vendor/zlib/gzlib.c:14:17: note: expanded from macro 'LSEEK'
#  define LSEEK lseek
                ^
../vendor/zlib/gzlib.c:402:15: error: implicit declaration of function 'lseek' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
        ret = LSEEK(state->fd, offset - (z_off64_t)state->x.have, SEEK_CUR);
              ^
../vendor/zlib/gzlib.c:14:17: note: expanded from macro 'LSEEK'
#  define LSEEK lseek
                ^
../vendor/zlib/gzlib.c:498:14: error: implicit declaration of function 'lseek' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
    offset = LSEEK(state->fd, 0, SEEK_CUR);
             ^
../vendor/zlib/gzlib.c:14:17: note: expanded from macro 'LSEEK'
#  define LSEEK lseek
                ^
5 errors generated.
make[1]: *** [obj/x64/Debug/zlib/gzlib.o] Error 1
make[1]: *** Waiting for unfinished jobs....
../vendor/zlib/gzread.c:35:15: error: implicit declaration of function 'read' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
        ret = read(state->fd, buf + *have, get);
              ^
../vendor/zlib/gzread.c:35:15: note: did you mean 'fread'?
/Applications/Xcode_13.2.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/stdio.h:158:9: note: 'fread' declared here
size_t   fread(void * __restrict __ptr, size_t __size, size_t __nitems, FILE * __restrict __stream);
         ^
../vendor/zlib/gzread.c:649:11: error: implicit declaration of function 'close' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
    ret = close(state->fd);
          ^
2 errors generated.
make[1]: *** [obj/x64/Debug/zlib/gzread.o] Error 1
make: *** [zlib] Error 2
Error: Process completed with exit code 2.
```

### PR DESCRIPTION
**Pull request by Lpsd**
an attempt to resolve #2585

### CODE REVIEW

**Comment by Lpsd** [comment]
first run failed, have tried adding a wildcard to system filter - waiting for build again.

**Comment by Lpsd** [comment]
all good to go 🎉

File: vendor/zlib/premake5.lua [hunk 1]
```diff
@@ -20,3 +20,6 @@ project "zlib"
 	}
 
 	defines { "verbose=-1" }    -- Stop "bit length overflow" warning
+
+	filter "system:Mac*"
+		defines { "Z_HAVE_UNISTD_H" }
```
Comment (patrikjuvonen) [inline vendor/zlib/premake5.lua]: Nice, I believe it should be `HAVE_UNISTD_H` to follow the [zlib configure source](https://github.com/madler/zlib/blob/cacf7f1d4e3d44d871b605da3b647f07d718623f/zconf.h.in#L434) options.

**Review by patrikjuvonen — CHANGES_REQUESTED** [review]

**Review by patrikjuvonen — DISMISSED** [review]

File: vendor/zlib/premake5.lua (hunk 1 above)
Comment (Lpsd) [reply vendor/zlib/premake5.lua]: you're right - I've updated this

**Review by patrikjuvonen — APPROVED** [review]

**Pull request merged**